### PR TITLE
Variations: First Variation -> Navigate to EditAttributeList after creating first attribute

### LINKED ISSUE #3536: Variations: Attribute List UI
After creating a new product attribute(#3535), the user should be redirected to this screen 

Screen Shot 2021-01-27 at 2 26 40 PM

Tasks:
- [ ] Build UI: Cell + Add attribute button + Done button CTA
- [ ] Connect "Next" CTA(after updating the product) to this screen

### LINKED ISSUE #3618: Variations: Encapsulate new/edit attribute flow
Think about a lightweight coordinator object to handle the add/edit attribute flow. 

It should: 
- [ ] Start on the `AddAttributeViewController`
- [ ] End on the `AttributeListViewController`
- [ ] Provide a completion block when the block is finished

Needs to be spawed from:
- [ ] Variation list (Empty state and edit attributes)
- [ ] Attribute List

### PR DESCRIPTION
**Pull request by Ecarrion**
closes #3536  

# Why

Creating the first variation is a big and complex flow, for that reason, I'm splitting it into several smaller flows.
Today's turn will be to: **Navigate to "Edit Attribute List" after creating the first attribute**, this will allow merchants to add more attributes before generating the first variation.

Note: This PR, does not implement any functionality inside the `EditAttribute` screen, this is only the UI, real content will be handled in #3537 

# How

- Adds `EditAttributesViewController` with dummy content.

- Update `ProductVariationsViewController` to present `EditAttributesViewController` after finishing creating/updating the first attribute. A note here, is that the view controllers used to create the attribute(`AddAttributeVC` and `AddAttributeOptionVC`) are being removed before presenting the `EditAttributesVC`.

# Screenshots

Light | Dark
--- | ---
normal | dark-mode

Landscape | Big Font
--- | ---
landscape | big-font

# Demo

demo

# Testing Steps
- Navigate to a variable product with no variations added
- Tap both of the "Add variations" buttons
- Tap on an existing attribute or write a name for a new one.
- Tap one(or more) existing options or add new ones by writing their name on the text field.
- Tap on the "Next" button
- When the loading finishes, see that you are redirected to a dummy attribute list screen.
- When tapping back, you should be redirected back to the empty state screen(not the Add Attribute Option Screen)

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Variations/Edit Attributes/EditAttributesViewController.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        configureRightButton()
+    }
+
+    func configureTitle() {
+        title = Localization.title
+    }
+
+    func configureRightButton() {
+        let rightBarButton = UIBarButtonItem(title: Localization.done,
+                                             style: .done,
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Products/Variations/Edit Attributes/EditAttributesViewController.swift]: Maybe switch to `barButtonSystemItem: .done` to reuse system styling and localizations?

**Review by ealeksandrov — CHANGES_REQUESTED** [review]
Code and new UI looks great 👍

I see the only issue with navigation when you trigger next and then go back immediately:

https://user-images.githubusercontent.com/3132438/107501048-f93a7380-6ba7-11eb-975c-2659203ee531.mp4

What do you think about it? Should we cancel the network operation? Or forward navigation? Or maybe this is too far of edge case? App works well anyway, nothing really breaks 🙂

**Comment by Ecarrion** [comment]
Hey @ealeksandrov, thanks for the review! I've updated the bar button item as per your suggestion.

Regarding the back button problem, I have [this card](https://github.com/woocommerce/woocommerce-ios/projects/29#card-54557890) to not forget about that edge case because I think it's better to deal with it outside of these PR's

Would you mind converting that card into an issue, explain the problem, and provide possible alternatives so the design team can weigh in?

Thanks!

**Review by ealeksandrov — APPROVED** [review]
Agree! LGTM 🚀

**Pull request merged**